src/cider/type/plan/BinaryExpr.cpp

```diff
@@ -164,11 +247,20 @@ JITExprValue& BinOper::codegenFixedSizeColCmpFun(JITValuePointer& null,
 JITExprValue& BinOper::codegenFixedSizeLogicalFun(JITFunction& func,
                                                   JITValuePointer& null,
                                                   FixSizeJITExprValue& lhs_val,
-                                                  FixSizeJITExprValue& rhs_val) {
+                                                  FixSizeJITExprValue& rhs_val,
+                                                  bool branchless_logic) {
   switch (get_optype()) {
     case kAND: {
       // If one side is not null and false, return not null and FALSE
       // else no special handle needed
+      if (branchless_logic) {
+        auto lhs_null = lhs_val.getNull();
+        auto lhs_data = lhs_val.getValue();
+        auto rhs_null = rhs_val.getNull();
+        auto rhs_data = rhs_val.getValue();
+        null = null || (lhs_null && lhs_data) || (rhs_null && rhs_data);
+        return set_expr_value(null, lhs_data && rhs_data);
+      }
```

Comment:
The branchless code was existing but commented. But seems your logic is not the same, can you take a look at?

Reply:
fix it for the first branch, it's `&&`, not `||`